ui/components/MeshModelComponent.js

```diff
@@ -1,23 +1,211 @@
+import { getComponentsDetail, getModelsDetail } from '../api/meshmodel'
+
 const meshmodelStyles = (theme) => ({
-  wrapperClass : {
-    padding : theme.spacing(2),backgroundColor : theme.palette.secondary.elevatedComponents,height : "60vh",marginTop : theme.spacing(2),borderRadius : theme.spacing(1),
+  wrapperClss : {
+    flexGrow : 1,
+    maxWidth : '100%',
+    height : 'auto',
+  },
+  tab : {
+    minWidth : 40,
+    paddingLeft : 0,
+    paddingRight : 0,
+    "&.Mui-selected" : {
+      color : theme.palette.type === 'dark' ? "#00B39F" : theme.palette.primary,
+    },
+  },
+  tabs : {
+    "& .MuiTabs-indicator" : {
+      backgroundColor : theme.palette.type === 'dark' ? "#00B39F" : theme.palette.primary,
+    },
   },
   dashboardSection : {
-    padding : theme.spacing(2),borderRadius : 4,height : "100%",overflowY : "scroll"
+    padding : theme.spacing(2), borderRadius : 4,height : "100%",overflowY : "scroll"
   },
 
+
 })
-const MeshModelComponent = ({ showMeshModelSummary,classes }) => {
+
+const MeshModelComponent = ({ type, classes }) => {
+  const [resourcesDetail, setResourcesDetail] = useState();
+  const [isRequestCancelled, setRequestCancelled] = useState(false);
+  const [page, setPage] = useState(1);
+
+  const getModels = async (page) => {
+    try {
+      const data = await getModelsDetail(page);
+      if (!isRequestCancelled) {
+        setResourcesDetail(data);
+      }
+    } catch (error) {
+      console.error('Failed to fetch models:', error);
+    }
+  };
+
+  const getComponents = async (page) => {
+    try {
+      const data = await getComponentsDetail(page);
+      if (!isRequestCancelled) {
+        setResourcesDetail(data);
+      }
+    } catch (error) {
+      console.error('Failed to fetch components:', error);
+    }
+  };
+
+  useEffect(() => {
+
+    setRequestCancelled(false);
+
+
+    if (type === 'models') {
+      getModels(page);
+    } else if (type === 'components') {
+      getComponents(page);
+    }
+
+    return () => {
+      setRequestCancelled(true);
+    };
+  }, [type, page]);
+
+
+
+
+  const smi_columns = [
+    {
+      name : type === 'components' ? 'kind' : 'displayName',
+      label : `Name`,
+      options : {
+        sort : true,
+        sortDescFirst : true,
+        sortThirdClickReset : true,
+        customHeadRender : function CustomHead({ index, ...column }, sortColumn) {
+          return (
+            <TableCell align={"start"} key={index} onClick={() => sortColumn(index)}>
+              <TableSortLabel active={column.sortDirection != null} direction={column.sortDirection || "asc"}>
+                <b>{column.label}</b>
+              </TableSortLabel>
+            </TableCell>
+          );
+        },
+        customBodyRender : (value) => (
+          <Tooltip title={value} placement="top">
+            <div>{value}</div>
+          </Tooltip>
+        )
+      },
+    },
+    {
+      name : type === 'components' ? 'apiVersion' : 'version',
```

Comment:
`components` you mean `component-view`?

Reply:
It means if a user is viewing components or models

Reply:
okay, lets give it the appropriate name then.

Reply:
How about view === "components" or view === "models" ?